addons/web/static/src/js/core/ajax.js

```diff
@@ -433,15 +433,65 @@ var loadXML = (function () {
     };
 })();
 
-return {
+
+/**
+ * Makes sure that the jsLibs and cssLibs are properly loaded. Note that
+ * the ajax loadJS and loadCSS methods don't do anything if the given file
+ * is already loaded.
+ *
+ * @param {Object} libs
+ * @param {Array<string> | Array<Array<string>>} [libs.jsLibs=[]] The list of JS files
+ *   that we want to load.  If it is a list of strings, each file will be loaded
+ *   in parallel.  If it is a list of lists, the inner lists will be loaded
+ *   sequentially, and each element of the main list will be loaded in parallel.
+ * @param {Array<string>} [libs.cssLibs=[]] A list of css files, to be loaded in
+ *   parallel
+ *
+ * @returns {Deferred}
+ */
+function loadLibs (libs) {
+    var defs = [];
+    _.each(libs.jsLibs || [], function (urls) {
+        if (typeof(urls) === 'string') {
+            // jsLibs is an array of urls: those urls can be loaded in
+            // parallel
+            defs.push(ajax.loadJS(urls));
+        } else {
+            // jsLibs is an array of arrays of urls: those arrays of urls
+            // can be loaded in parallel, but the urls inside each
+            // sub-array must be loaded sequentially
```

Comment:
why did you changed the semantics?

Reply:
It was very strange that the behavior changed depending on the content of the list. For if it was a simple list it was in parallel, while for a list of list it became sequential with the inside in parallel. So if we are a widget by adding libs, this can change the order of loading. Moreover, the most frequent and sequential case